### PR TITLE
nekrs: add v23.0, add new build system

### PR DESCRIPTION
**Pull request by snehring**
They've switched to cmake.

I'm not actually able to successfully test the old version as it won't even compile for me with the original package recipe, but I believe we should be in the same place there with this change.

### CODE REVIEW

File: var/spack/repos/builtin/packages/nekrs/package.py [hunk 1]
```diff
@@ -111,26 +135,14 @@ def setup_build_environment(self, env):
...
-
-        with working_dir(script_dir):
-            # Make sure nekmpi wrapper uses srun when we know OpenMPI
-            # is not built with mpiexec
-            if "^openmpi~legacylaunchers" in spec:
-                filter_file(r"mpirun -np", "srun -n", "nrsmpi")
-                filter_file(r"mpirun -np", "srun -n", "nrspre")
-                filter_file(r"mpirun -np", "srun -n", "nrsbmpi")
 
+class GenericBuilder(spack.build_systems.generic.GenericBuilder):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nekrs/package.py:146]: Missing:
```python
import spack.build_systems
```
?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just to be double check, did you try compiling the latest version?

File: var/spack/repos/builtin/packages/nekrs/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/nekrs/package.py:146]: I had, several times. Confused as to why that works for me. Lemme see if I can reproduce after cleaning.

Comment (snehring) [reply var/spack/repos/builtin/packages/nekrs/package.py:146]: Yeah, it still works. Weird. I'll add explicit imports.

**Comment by alalazo** [comment]
@spackbot maintainers

**Review by alalazo — APPROVED** [review]

**Pull request merged**